resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/internal/OpenToHalfOpenAutoTransitioner.java

```diff
@@ -0,0 +1,56 @@
+package io.github.resilience4j.circuitbreaker.internal;
+
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+
+import java.time.Duration;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Enables the the circuit breaker to automatically transition from Open To Half Open state after a time has passed.
+ * Used when enableAutomaticTransitionFromOpenToHalfOpen config property is set to true.
+ */
+public class OpenToHalfOpenAutoTransitioner {
+
+    private OpenToHalfOpenAutoTransitioner() {
+    }
+
+    public static void scheduleAutoTransitionToHalfOpen(CircuitBreaker circuitBreaker) {
+        if (circuitBreaker.getRetryAfterWaitDuration().isEmpty()
+                || !circuitBreaker.getState().equals(CircuitBreaker.State.OPEN)) {
+            return;
+        }
+
+        ScheduledExecutorService executorService = LazyScheduledExecutorService.getExecutorService();
```

Comment:
Here you could use a feature of Vavr.
`ScheduledExecutorService executorService  = Lazy.of(Executors::newSingleThreadScheduledExecutor).get();`
You could remove the class `LazyScheduledExecutorService` then.